### PR TITLE
Adds workflow to publish site

### PR DESCRIPTION
**Pull request by bdemers**

### CODE REVIEW

**Comment by bdemers** [comment]
Once we are happy with manually triggering this, we can enable it to run when changes are merged to `main`

**Review by lprimak — APPROVED** [review]

**Review by fpapon — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      - name: 'Checkout target'
+        uses: actions/checkout@v4
+        with:
+          repository: 'apache/shiro-site'
+          ref: 'asf-site'
+          path: 'asf-site'
+
+      - name: 'Copy build to site repo'
+        run: |
+          rsync -a src/target/website/. ./asf-site
```
Comment (bmarwell) [inline .github/workflows/publish.yml:53]: What about deleting files?

Comment (bmarwell) [reply .github/workflows/publish.yml:53]: I think we should append ```
--delete
```

Comment (bdemers) [reply .github/workflows/publish.yml:53]: I'm not sure, that _could_ cause 404's, but that might be better than orphaned files?

Comment (lprimak) [reply .github/workflows/publish.yml:53]: I think Ben is correct. The automated flow needs to be the same as the [manual one](https://github.com/apache/shiro-site/blob/main/README.adoc) and needs `-avP --delete --exclude .git` cmd line arguments

Comment (bdemers) [reply .github/workflows/publish.yml]: Ahh! It's in the doc, so I'll update the script.

Longer term we may want to think about how we prevent 404's but... it hasn't been an issue so I don't think we need to worry about it right now :D

File: .github/workflows/staging.yml [hunk 2]
```diff
@@ -59,7 +59,7 @@ jobs:
 
       - name: 'Copy to staging'
         run: |
-          rsync -a src/target/website/. ./site-staging 
+          rsync -avP --delete --exclude .git src/target/website/. ./asf-staging
```
Comment (bdemers) [inline .github/workflows/staging.yml:62]: I updated the staging.yml to match use the same rsync command

**Review by lprimak — APPROVED** [review]

**Pull request merged**